src/api/l_physics_world.c

```diff
@@ -223,15 +219,15 @@ static int l_lovrWorldSetSleepingAllowed(lua_State* L) {
 
 static int l_lovrWorldRaycast(lua_State* L) {
   World* world = luax_checktype(L, 1, World);
-  float x1 = luax_checkfloat(L, 2);
-  float y1 = luax_checkfloat(L, 3);
-  float z1 = luax_checkfloat(L, 4);
-  float x2 = luax_checkfloat(L, 5);
-  float y2 = luax_checkfloat(L, 6);
-  float z2 = luax_checkfloat(L, 7);
-  luaL_checktype(L, 8, LUA_TFUNCTION);
-  lua_settop(L, 8);
-  lovrWorldRaycast(world, x1, y1, z1, x2, y2, z2, raycastCallback, L);
+  float start[4], end[4];
+  int index;
+  index = luax_readvec3(L, 2, start, NULL);
+  index = luax_readvec3(L, index, end, NULL);
+  luaL_checktype(L, index, LUA_TFUNCTION);
+  lua_settop(L, index);
+  lovrWorldRaycast(world, 
```

Comment:
Minor style: I'd prefer to keep this unwrapped since it's still around 100 characters.